action/action-simulator/src/main/java/com/powsybl/action/simulator/loadflow/LoadFlowActionSimulator.java

```diff
@@ -401,4 +408,56 @@ private LoadFlowResult runTest(RunningContext context, Network networkForTry, Ac
             throw new PowsyblException(e);
         }
     }
+
+    private interface NetworkCopyStrategy {
+
+        void clearState();
+
+        Network restoreState(String stateId);
+
+        void saveState();
+    }
+
+    private class DeepCopyStrategy implements NetworkCopyStrategy {
+        private byte[] contextNetwork;
+
+        @Override
+        public void clearState() {
+            // Nothing to do
+        }
+
+        @Override
+        public Network restoreState(String id) {
+            return NetworkXml.gunzip(contextNetwork);
+        }
+
+        @Override
+        public void saveState() {
+            contextNetwork = NetworkXml.gzip(network);
```

Comment:
It seems that `NetworkCopyStrategy` instances are always using the `network` instance. Shouldn't we use the network instance if the EvaluationContext, for instance in the `runTests`. Have you really tested this use-case?